src/components/Account/Account.tsx

```diff
@@ -46,6 +46,8 @@ const Account = ({ panelClassName, panelHeaderClassName, canUpdateEmail = true }
   const exportData = useMutation(exportAccountData);
   const [isAlertVisible, setIsAlertVisible] = useState(false);
   const exportDataMessage = exportData.isSuccess ? t('account.export_data_success') : t('account.export_data_error');
+  const accountData = useAccountStore();
+  const shouldAddPassword = accountData.user?.metadata?.register_source !== 'inplayer' && accountData.user?.metadata?.has_password !== 'true';
```

Comment:
move this below useAccountStore definition:
`const userHasPassword = user?.metadata?.register_source !== 'inplayer' && !user?.metadata?.has_password;`

Reply:
But it would be better if this is moved on service level, this way it will probably interfere with Cleeng's logic.